altapay.php

```diff
@@ -0,0 +1,2353 @@
+     * @return mixed
+     */
+    public function hookBackOfficeHeader($params)
+    {
+        $cookie = $this->context->cookie;
+
+        $this->context->controller->addJquery();
+        $this->context->controller->addJS($this->_path . 'views/js/form.js', 'all');
+
+        if ($cookie->altapayError) {
+            $this->context->controller->errors[] = $cookie->altapayError;
+
+            // unset the variable:
+            $cookie->altapayError = null;
+        }
+
+        // always returns false because there is no template to display
+        return false;
+    }
+
+    /**
+     * Captures a payment when the status is changed to Shipped
+     *
+     * @param $params
+     * @return mixed
+     */
+    public function hookActionOrderStatusUpdate($params)
+    {
+        $results = $this->selectOrder($params);
+
+        if (!$results) {
+            return;
+        }
+
+        $paymentID = $results['payment_id'];
+
+        /** @var OrderStateCore */
+        $newStatus = $params['newOrderStatus'];
+
+        $shippedStatus = Configuration::get('PS_OS_SHIPPING');
+
+        if (!empty($newStatus)) {
+            if ($newStatus->id == $shippedStatus) { // a capture will be made if necessary
+                $this->performCapture($paymentID, $params);
+            }
+        } else {
+            return;
+        }
+        return $results;
+    }
+
+    /**
+     * @param $params
+     * @return mixed
+     */
+    private function selectOrder($params)
+    {
+        return Db::getInstance()->getRow('SELECT ' . _DB_PREFIX_ . 'altapay_order.*, '
+            . _DB_PREFIX_ . 'altapay_transaction.amount FROM `'
+            . _DB_PREFIX_ . 'altapay_order` INNER JOIN ' . _DB_PREFIX_ . 'altapay_transaction ON '
+            . _DB_PREFIX_ . 'altapay_transaction.unique_id = ' . _DB_PREFIX_ . 'altapay_order.unique_id 
+        WHERE id_order=' . $params['id_order']);
+    }
+
+    /**
+     * Method is being triggered whenever capture action is performed
+     * @param $paymentID
+     * @param $params
+     * @param bool $captureRemainedAmount
+     */
+    public function performCapture($paymentID, $params, $captureRemainedAmount = true)
+    {
+        try {
+            $api = new MerchantAPI();
+            $productDetails = new OrderDetail;
+            $api->init($this->getAltapayUrl(), $this->getAPIUsername(), $this->getAPIPassword());
+            $paymentDetails = $api->getPaymentDetails($paymentID);
+            $orderReservedAmount = $paymentDetails->getReservedAmount();
+            $orderCapturedAmount = $paymentDetails->getCapturedAmount();
+            $amountToCapture = $orderReservedAmount - $orderCapturedAmount;
+
+            $giftWrappingFee = null;
+
+            if ($productDetails->gift) {
+                $giftWrappingFee = $productDetails->total_wrapping;
+            }
+
+            if ($amountToCapture == 0) {
+                return;
+            } elseif ($amountToCapture > 0 && $orderCapturedAmount == 0) {
```

Comment:
previous statment returns, no need for `elseif`
```suggestion
            }
            if ($amountToCapture > 0 && $orderCapturedAmount == 0) {
```